src/WorkflowCore/Models/ExecutionPointer.cs

```diff
@@ -45,6 +45,8 @@ public class ExecutionPointer
         public object Outcome { get; set; }
 
         public PointerStatus Status { get; set; } = PointerStatus.Legacy;
+        
+        public Exception CurrentException { get; set; }
```

Comment:
This class needs to be serializable... not sure how different sub-classes of `Exception` will behave in that case?

Reply:
As far as I understood, error handling happens in the same persistence round (not sure if this is how it's called tho) when the exception occurs. So do we really need to serialize this field?

Reply:
Why do we need this field on the `ExecutionPointer`?

Reply:
@danielgerlag We need this field to build `StepExecutionContext` so that we can retrieve a thrown exception in `Catch()` step. An example:
```c#
builder.Try(b => b.StartWith(_ => throw new ArgumentException("I am Exception1")))
            .Catch(new[] {typeof(Exception)}, ctx =>
            {
                Console.WriteLine($"Caught an exception: Type: '{ctx.CurrentException.GetType().Name}', Message: '{ctx.CurrentException.Message}'");
                return ExecutionResult.Next();
            })
```